lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -2391,20 +2393,33 @@ void BoundInterpreterFunction::fwdSparseLengthsWeightedSumGradInst(
 
   auto IGH = destGrad->getHandle();
   auto WH = weights->getHandle();
+  auto WGH = weightsGrad->getHandle();
+  auto DH = data->getHandle();
   auto OGH = dataGrad->getHandle();
 
-  // For each index in each segment, accumulate into the corresponding data
-  // gradient the product of the gradient of the result it was added to and
-  // the weight that it was multiplied by during the
-  // SparseLengthsWeightedSum operation.
+  // For each index in each segment:
+  //    1) accumulate into the corresponding data gradient the product of the
+  //    gradient of the result it was added to and the weight that it was
+  //    multiplied by during the SparseLengthsWeightedSum operation.
+  //
+  //    2) accumulate into each weight gradient the reduced sum of the
+  //    elementwise product of the result slice that the corresponding weight
+  //    produced and the input slice that the weight was multiplied with.
   size_t curIdx = 0;
   for (size_t i = 0; i < segments; i++) {
-    size_t offsetIn = i * lineSize;
+    size_t destOffset = i * lineSize;
     for (size_t j = 0, e = LH.raw(i); j < e; j++) {
```

Comment:
Matter of taste, usually in some cases Iterator++ vs ++Iterator - the prefix form is faster.